cliquet/resource.py

```diff
@@ -202,10 +188,16 @@ def raise_412_if_modified(self, record=None):
                         error=HTTPPreconditionFailed.title,
                         message=error_msg),
                     content_type='application/json')
-                self.add_timestamp_header(response)
+                self._add_timestamp_header(response)
                 raise response
 
-    def raise_conflict(self, exception):
+    def _raise_invalid(self, location='body', **kwargs):
+        """Helper to raise a validation error."""
+        self.request.errors.add(location, **kwargs)
+        response = errors.json_error(self.request.errors)
+        raise response
+
+    def _raise_conflict(self, exception):
```

Comment:
Should this stay public, in case a library user would like to add it to other endpoints?